ui/lib/core/addon/components/form-field.js

```diff
@@ -193,9 +195,13 @@ export default class FormFieldComponent extends Component {
 
   @action
   handleChecklist(event) {
-    const valueArray = this.args.model[this.valuePath];
-    const method = event.target.checked ? 'addObject' : 'removeObject';
-    valueArray[method](event.target.value);
-    this.setAndBroadcast(valueArray);
+    let updatedValue = this.args.model[this.valuePath];
+    if (event.target.checked) {
+      updatedValue = addToArray(updatedValue, event.target.value);
+    } else {
+      updatedValue = removeFromArray(updatedValue, event.target.value);
+    }
+    this.args.model[this.valuePath] = updatedValue;
```

Comment:
I think `this.setAndBroadcast` handles setting the model, so I don't think line 204 is necessary?
```js
  @action
  setAndBroadcast(value) {
    this.args.model.set(this.valuePath, value);
    this.onChange(this.valuePath, value);
  }

Reply:
I'm matching what the previous line 198 does (either `valueArray.addObject(value)` or `valueArray.removeObject(value)`. It might be unnecessary but I'm hesitant to change such a far-reaching behavior here

Reply:
🤔 Ah, to me it looks like lines 200 and 202 are adding or removing the object. I don't think 198 `valueArray[method](event.target.value);` updated the original model value, that appeared left to line 199, which is why the refactor appeared to duplicate that process.